openedx/core/djangoapps/credit/signals.py

```diff
@@ -5,15 +5,79 @@
 
 import logging
 
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ObjectDoesNotExist
 from django.dispatch import receiver
 from django.utils import timezone
 from opaque_keys.edx.keys import CourseKey
-
+from openedx_events.learning.signals import (
+    EXAM_ATTEMPT_ERRORED,
+    EXAM_ATTEMPT_REJECTED,
+    EXAM_ATTEMPT_RESET,
+    EXAM_ATTEMPT_SUBMITTED,
+    EXAM_ATTEMPT_VERIFIED
+)
+
+from openedx.core.djangoapps.credit.api.eligibility import (
+    is_credit_course,
+    remove_credit_requirement_status,
+    set_credit_requirement_status
+)
 from openedx.core.djangoapps.signals.signals import COURSE_GRADE_CHANGED
 
+User = get_user_model()
+
 log = logging.getLogger(__name__)
 
 
+def handle_exam_event(signal, event_data, credit_status=None):
+    """
+    update credit requirements based on exam event
+    """
+    user_data = event_data.student_user
+    course_key = event_data.course_key
+    usage_key = event_data.usage_key
+    request_namespace = 'exam'
+
+    # quick exit, if course is not credit enabled
+    if not is_credit_course(course_key):
```

Comment:
More of a hypothetical long term question here, but is there any reason that we might _possibly_ want a log here, e.g. for tracking how many times we're sending event signals for non-credit courses for some reason?

Reply:
IMO don't see a ton of value logging here. The event bus itself should be observable (somehow) for what's emitted.